plugins/modules/ec2_security_group.py

```diff
@@ -842,12 +842,14 @@ def ports_expand(ports):
     # takes a list of ports and returns a list of (port_from, port_to)
     ports_expanded = []
     for port in ports:
-        if not isinstance(port, string_types):
-            ports_expanded.append((port,) * 2)
-        elif '-' in port:
-            ports_expanded.append(tuple(int(p.strip()) for p in port.split('-', 1)))
-        else:
+        try:
             ports_expanded.append((int(port.strip()),) * 2)
+        except ValueError as e:
```

Comment:
May I suggest to isolate this block in a dedicated function with an unit-test?

Reply:
Totally agree, and plan on picking things apart further (hence #1212 is still open).  Still working on wrapping my head around some of the "expansion" code